### DISCUSSION TITLE
Why text sensor instead of binary sensor for zones?

### DISCUSSION THREAD

**Discussion by AlexandreUSA** [Dilbert66/esphome-vistaECP#37] [Q&A] [answered]
What is the reason for the text_sensors instead of binary_sensors for zones. Is it because the backend code requires strings? Binary sensor in HA would let me see the logbook straight from clicking a sensor, change the icon base on the sensor state (O/C). Are they not always C or O?

**Comment by Dilbert66** [comment, marked as answer]
Zones can be "[O]pen, [C]losed,[B]ypassed,[A]larm,'[T]ampered.  This way with one field I can show the state that a zone is in, instead of having multiple binary sensors per zone.  If you had 32 zones, you would need 128 fields to show all possible statuses.   An alternate way perhaps is to use a binary sensor for each zone showing open,close then use a large text field to show which zones are bypassed or in alarm for example. B1,B2,B3,  or A1,A2,A3  .   I chose to have a multi purpose text field to keep it simple.   I realize, it's not easy to associate an icon to it due to it being non binary.    

You could modify the yaml to make everything into binary sensors if you preferred and just used the O or C statuses but for my own use I prefer not to.

**Reply by AlexandreUSA** [reply]
Nice, I did not realize that there were 5 states associated to the zone. Just tested the bypass, and indeed, I got a B next to that zone.

By the way, interesting that the message "bypass zone restored" is when you bypass the zone and not when you unbypass.  
![Bypass](https://user-images.githubusercontent.com/56739736/121412462-be1c3700-c932-11eb-8da2-c646c3923e0a.JPG)

**Comment by Dilbert66** [comment]
Some of the LRR messages need to have special criteria to indicate state. I've added a specific case for bypass zones to show the correct state. I've pushed the change to master.  Only vistaalarm.h needs to be updated

**Reply by AlexandreUSA** [reply]
This is great. Love the active vs cleared. 

There is something not right with the HA log. I don't think it is an issue on the vistaalarm. 
When I bypass, the bypass zone is now active  

4:25pm: bypass zone 5 active - all normal here
![log1](https://user-images.githubusercontent.com/56739736/121424622-88ca1600-c93f-11eb-820f-b17076eaf7cc.JPG)

4:28pm: bypass zone 5 cleared - still all good 
![log2](https://user-images.githubusercontent.com/56739736/121425009-f413e800-c93f-11eb-8f8d-4c862262c368.JPG)

A minute later, I am now missing the entry in the log for the active one at 4:25pm. I have the same issue with zones with missing entries. Weird.
![log3](https://user-images.githubusercontent.com/56739736/121425273-3806ed00-c940-11eb-94df-901adb1619e9.JPG)
